vm_supervisor/run.py

```diff
@@ -145,7 +145,7 @@ async def run_code_on_request(
         headers.update(
             {
                 "Aleph-Program-ItemHash": execution.vm_hash,
-                "Aleph-Program-Code-Ref": execution.program.code.ref,
+                "Aleph-Program-Code-Ref": execution.message.code.ref if not execution.is_instance else None,
```

Comment:
Why this change ?

Reply:
Because an instance don't have a code reference, because don't have any code.